src/Service.Tests/SqlTests/RestApiTests/Put/MsSqlPutApiTests.cs

```diff
@@ -364,6 +382,83 @@ public virtual async Task PutOneWithRowversionFieldInRequestBody()
                 expectedSubStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest.ToString()
                 );
         }
+        /// <summary>
+        /// Test to validate that even when an update DML trigger is enabled on a table, we still return the
+        /// latest data as it is present after the trigger gets executed. Whenever an update DML trigger is enabled,
+        /// we use a subsequent SELECT query to get the data instead of using OUTPUT clause. To validate that the data is returned
+        /// as it is after the trigger is executed, we use the values which are updated by the trigger in the WHERE predicates of the verifying sql query.
+        /// </summary>
+        [TestMethod]
+        public async Task PutOneUpdateInTableWithUpdateTrigger()
```

Comment:
refer to https://github.com/Azure/data-api-builder/pull/1630#discussion_r1325210147